python/pyspark/sql/connect/session.py

```diff
@@ -827,8 +827,16 @@ def stop(self) -> None:
         # other remote clients being used from other users.
         with SparkSession._lock:
             if not self.is_stopped and self.release_session_on_close:
-                self.client.release_session()
-            self.client.close()
+                try:
```

Comment:
Please add some documentation on why we're doing this - e.g. to recover easily from network issues and not requiring the final RPC call to be successful because stop() is the last call of the session.